core/src/main/java/org/apache/carbondata/core/dictionary/generator/ServerDictionaryGenerator.java

```diff
@@ -64,24 +66,34 @@ public void initializeGeneratorForTable(DictionaryMessage key) {
     }
   }
 
+  public void initializeGeneratorForColumn(DictionaryMessage key) {
+    tableMap.get(key.getTableUniqueId()).updateGenerator(key);
```

Comment:
make sure get Method is not returning null from the map else NullPointerException will be thrown